chain/store/store.go

```diff
@@ -576,10 +656,35 @@ func (cs *ChainStore) GetCurrentBlockStateRoot() (common.Uint256, error) {
 	return hash, nil
 }
 
+func (cs *ChainStore) GetFastSyncStateRoot() (common.Uint256, error) {
+	currentState, err := cs.st.Get(db.CurrentFastSyncRoot())
+	if err != nil {
+		return common.EmptyUint256, err
+	}
+
+	hash, err := common.Uint256ParseFromBytes(currentState)
+	if err != nil {
+		return common.EmptyUint256, err
+	}
+
+	return hash, nil
+}
+
 func (cs *ChainStore) GetDatabase() db.IStore {
 	return cs.st
 }
 
+func (cs *ChainStore) DuringFastSync(height uint32) bool {
```

Comment:
Suggest rename to `IsDuringFastSync` and only use this function once at the block sync logic, and pass it along the call just like `pruning`